framework/include/utils/ParallelUniqueId.h

```diff
@@ -32,20 +37,30 @@ class ParallelUniqueId
   ParallelUniqueId()
   {
 #ifdef LIBMESH_HAVE_TBB_API
-    ids.pop(id);
-#else
-#ifdef LIBMESH_HAVE_PTHREAD
-    id = Threads::pthread_unique_id();
+    _ids.pop(id);
+#elif LIBMESH_HAVE_OPENMP
+    id = omp_get_thread_num();
+#elif LIBMESH_HAVE_PTHREAD
+    Threads::spin_mutex::scoped_lock lock(_pthread_id_mutex);
+
+    if (_ids.empty())
+      throw MooseException("No Thread IDs available in ParallelUniqueID. Did you forget to initialize()?");
+
+    id = _ids.front();
+    _ids.pop();
```

Comment:
Right but you still can't create it more than once per thread. That's actually a really good point and would be present in the current version as well with TBB. What will happen with this implementation is you'll get an error which is nice. In TBB, the pop method will wait so it'll probably run but with potentially undefined behavior - Yuck!